core/src/main/java/jenkins/views/Header.java

```diff
@@ -0,0 +1,36 @@
+package jenkins.views;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.kohsuke.accmod.Restricted;
+import org.kohsuke.accmod.restrictions.NoExternalUse;
+
+import edu.umd.cs.findbugs.annotations.CheckForNull;
+import hudson.ExtensionList;
+import hudson.ExtensionPoint;
+
+/**
+ * Extension point that provides capabilities to render a specific header
+ * 
+ * @author Ildefonso Montero
+ * @see JenkinsHeader
+ */
+public abstract class Header implements ExtensionPoint {
```

Comment:
`interface` cleaner I think.

Reply:
Changed from `interface` to `abstract class` to apply: https://github.com/jenkinsci/jenkins/pull/5909#discussion_r749171416